testdata/p4_16_errors_outputs/issue-2123_e.p4-stderr

```diff
@@ -0,0 +1,2 @@
+error: 2054-2048: Range end is less than start.
```

Comment:
Is there a way to make the ranges appear the same as they did in the source code?  No big deal if not, but thought I would ask.  I know that the source code could have it appear in any base that P4_16 supports, so this would involve some compiler internal to access the source code form of the value.

Reply:
In the p4c code, it's a cinch to find the base of the constant defined in P4 code.  However, I am using p4c's `::error` API for which I don't know how to provide `std::hex` and `std::showbase` to print the constant in hex.

Reply:
I just noticed that there is no location within the source code associated with this error message.  A file name and line number would be good, yes?

Reply:
Totally agree.  The error reporting API I am using does not print the information.  Let me take a look.

Reply:
I am using the same `::error` API in midend code as the bmv2 backend simple_switch code.  The backend code, on hitting the error, does print the line number of P4 code.  However, my midend p4c code does not.  I will look into this one.  

The log below is from a p4c build without this PR's changes.  See that line numbers are printed,

```
hemant@ubuntu:~/p4c/build$ ./p4c-bm2-ss --std p4-16 /tmp/p4c/testdata/p4_16_errors/issue-2123_e.p4 
/tmp/p4c/testdata/p4_16_errors/issue-2123_e.p4(61): [--Werror=invalid] error: ..: Invalid must evaluate to a compile-time constant
            0x0806 .. 0x0800 : parse_ipv4;
            ^^^^^^^^^^^^^^^^
/tmp/p4c/testdata/p4_16_errors/issue-2123_e.p4(62): [--Werror=invalid] error: ..: Invalid must evaluate to a compile-time constant
            2054 .. 2048 : parse_ipv4;
            ^^^^^^^^^^^^
hemant@ubuntu:~/p4c/build$ 

```

Printing in same base is fixed by the new checkin.

Reply:
A `IR::Node` needs to be included in the error message and then the source file and line number are printed.  I checked in code.